packages/thrift-server-core/src/main/protocols/JSONProtocol.ts

```diff
@@ -0,0 +1,627 @@
+            this.tstack.splice(p + 1, 1)
+            this.tstack[p].push(tmpVal)
+        }
+
+        this.tstack[p] = '[' + this.tstack[p].join(',') + ']'
+    }
+
+    public writeBool(value: boolean) {
+        this.tstack.push(value ? 1 : 0)
+    }
+
+    public writeByte(i8: number) {
+        this.tstack.push(i8)
+    }
+
+    public writeI16(i16: number) {
+        this.tstack.push(i16)
+    }
+
+    public writeI32(i32: number) {
+        this.tstack.push(i32)
+    }
+
+    public writeI64(i64: number | Int64) {
+        if (i64 instanceof Int64) {
+            this.tstack.push(i64.toDecimalString())
+        } else {
+            this.tstack.push(i64)
+        }
+    }
+
+    public writeDouble(dbl: number) {
+        this.tstack.push(dbl)
+    }
+
+    public writeString(str: string | null) {
+        // We do not encode uri components for wire transfer:
+        if (str === null) {
+            this.tstack.push(null)
+        } else {
+            // concat may be slower than building a byte buffer
+            let escapedString = ''
+            for (let i = 0; i < str.length; i++) {
+                const ch = str.charAt(i) // a single double quote: "
+                if (ch === '"') {
+                    escapedString += '\\"' // write out as: \"
+                } else if (ch === '\\') {
+                    // a single backslash
+                    escapedString += '\\\\' // write out as double backslash
+                } else if (ch === '\b') {
+                    // a single backspace: invisible
+                    escapedString += '\\b' // write out as: \b"
+                } else if (ch === '\f') {
+                    // a single formfeed: invisible
+                    escapedString += '\\f' // write out as: \f"
+                } else if (ch === '\n') {
+                    // a single newline: invisible
+                    escapedString += '\\n' // write out as: \n"
+                } else if (ch === '\r') {
+                    // a single return: invisible
+                    escapedString += '\\r' // write out as: \r"
+                } else if (ch === '\t') {
+                    // a single tab: invisible
+                    escapedString += '\\t' // write out as: \t"
+                } else {
+                    escapedString += ch // Else it need not be escaped
+                }
+            }
+            this.tstack.push('"' + escapedString + '"')
+        }
+    }
+
+    public writeBinary(binary: string | Buffer) {
+        let str = ''
+        if (typeof binary === 'string') {
+            str = binary
+        } else if (binary instanceof Uint8Array) {
+            const arr = binary
+            for (const i of arr) {
+                str += String.fromCharCode(arr[i])
+            }
+        } else {
+            throw new TypeError(
+                'writeBinary only accepts String or Uint8Array.',
+            )
+        }
+        this.tstack.push('"' + btoa(str) + '"')
```

Comment:
`btoa` doesn't exist in Node. There's a reasonably well-used `npm` package that mimics the api: https://www.npmjs.com/package/btoa. Or you could do `Buffer.from(str).toString('base64')`.

Reply:
Good catch, went with the `Buffer` version and amended it into the original commit